firmware/chipignite/util/caravel/hk.py

```diff
@@ -0,0 +1,155 @@
+from .find_ftdi import find_ftdi
+from .gpio import Gpio
+from .defs import *
+from pyftdi.spi import SpiController
+import time
+
+
+class HKSpi:
+    def __init__(self, ftdi_device: str=None):
+        if ftdi_device == None:
+            ftdi_device = find_ftdi()
+        
+        self.spi = SpiController(cs_count=1)
```

Comment:
Just checking whether this `cs_count=1` is correct, given the scripts have otherwise been using `cs_count=2`. Not sure if there are any (un)intentional side-effects of the way it was done.

Reply:
I feel like there was a reason for this one, UART maybe? I was trying to get the UART working right with the board at the same time, that might have been me testing different configurations, then deciding that wasn't getting me anywhere and reverting it back to what I had thought was the original value. I have no idea, you could probably revert that one

Reply:
Thanks! [Tim's comment](https://github.com/efabless/caravel_board/pull/98#issuecomment-2198193036) indicates the count of 2 was a hangover from much older boards, and I understand this just would have made ADBUS4 a 2nd `/CS` (unused, where it otherwise is not connected on Rev 5A and below, and breaks out via J12 as a board GPIO in Rev 5B). So, I think your setting `cs_count=1` is correct.